### DISCUSSION TITLE
Will UWP be able to use a higher version of .Net?

### DISCUSSION THREAD

**Discussion by themerror** [dotnet/runtime#71298] [General] [answered]
.Net 5 claims to be a "Unified Platform", but today UWP does not support newer versions of .Net. While Microsoft has always recommended migrating UWP apps to WASDK, however,  UWP is the only choice for Xbox/Hololens/Teams development.

**Comment by hez2010** [comment, marked as answer]
We currently don't have a .NET Native compiler toolchain for .NET Core 3.1+.
Technically you can use any .NET version you like for UWP, but without .NET Native compilation you cannot submit your apps to Microsoft Store.

**Reply by AqlaSolutions** [reply]
This doesn't explain how exactly I can use for example .NET 7 in UWP app

**Reply by colejohnson66** [reply]
You can use .NET 7, but you can’t submit it to the Store.

**Reply by AqlaSolutions** [reply]
@colejohnson66 , but how to do it? Where is the switch?

**Reply by AqlaSolutions** [reply]
If I set `<TargetFrameworkVersion>v8.0</TargetFrameworkVersion>`, it doesn't build saying that `Microsoft\WindowsXaml\v17.0\Microsoft.Windows.UI.Xaml.CSharp.targets` is not found. Same for v7.0.

**Reply by Gaoyifei1011** [reply]
https://github.com/driver1998/NetCoreApp

You can visit this repository
